tests/robot-tests/tests/public_api/public_api_restricted.robot

```diff
@@ -0,0 +1,300 @@
+
+Upload third subject(large data file)
+    user waits until large data upload is completed
+    ...    ${SUBJECT_NAME_3}
+    ...    data-upload-import.csv
+    ...    data-upload-import.meta.csv
+
+Add data guidance to third subject
+    user clicks link    Data guidance
+    user enters text into data guidance data file content editor    ${SUBJECT_NAME_3}    meta content
+    user clicks button    Save guidance
+
+Create an API dataset through the amendment
+    user scrolls to the top of the page
+    user clicks link    API data sets
+    user waits until h2 is visible    API data sets
+
+    user clicks button    Create API data set
+    ${modal}=    user waits until modal is visible    Create a new API data set
+    user chooses select option    id:apiDataSetCreateForm-releaseFileId   ${SUBJECT_NAME_3}
+    user clicks button    Confirm new API data set
+
+    user waits until page finishes loading
+    user waits until modal is not visible    Create a new API data set    %{WAIT_LONG}
+
+User waits until the 2nd API dataset status changes to 'Failed'
+    user waits until h3 is visible    Draft version details
+    wait until keyword succeeds    20x    5s    Verify status of API Datasets    Failed
+
+Verify the contents inside the 'Draft API datasets' table
+    user clicks link    Back to API data sets
+    user waits until h3 is visible    Draft API data sets
+
+    user checks table column heading contains    1    1    Draft version    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    2    Name             xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    3    Status           xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    4    Actions          xpath://table[@data-testid='draft-api-data-sets']
+
+
+    user checks table cell contains    1    1    v1.0    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table cell contains    1    3    Failed    xpath://table[@data-testid='draft-api-data-sets']
+
+Add release note for new release amendment
+    user clicks link    Content
+    user adds a release note    Test release note two
+
+    ${date}    get current datetime    %-d %B %Y
+    user waits until element contains    css:#release-notes li:nth-of-type(1) time    ${date}
+    user waits until element contains    css:#release-notes li:nth-of-type(1) p    Test release note two
+
+Validate checklist error while API data set is still processing or being failed
+    user edits release status
+    user checks checklist errors contains
+    ...    1 issue that must be resolved before this release can be published.
+    user checks checklist errors contains either link
+    ...    All public API data set processing must be completed
+    ...    All failed public API data sets must be retried or removed
```

Comment:
I'm happy to leave this - I don't think there is any case here where we could miss a failure? - but I'd expect us to be checking for one then the other. Maybe the data set is processed so quickly and that's why you've done it this way?

Reply:
When processing large data files, two specific errors can occur depending on the speed of processing. Sometimes the files are processed quickly, while other times the process is slower. Also, there's an active [bug ticket](https://dfedigital.atlassian.net/browse/EES-5420) because large data files are consistently getting failing while processing. Given this situation, I decided to add some checks that can handle either outcome.

Reply:
Cool. If we will want to change this later when the devs have changed / fixed stuff, then you could add a comment here - but I'll leave it to you to decide what's best here.

Reply:
Added comments accordingly